File: v1/src/main/java/com/api/v1/creditcar/controller/AllCreditCardsRetrievalController.java
package com.api.v1.creditcar.controller;

import com.api.v1.creditcar.dtos.CreditCardResponseDto;
import com.api.v1.creditcar.services.AllCreditCardsRetrievalService;
import jakarta.validation.Valid;
import lombok.Getter;
import org.springframework.beans.factory.annotation.Autowired;
import org.springframework.http.HttpStatus;
import org.springframework.web.bind.annotation.*;
import reactor.core.publisher.Flux;
import reactor.core.publisher.Mono;

@RestController
@RequestMapping("api/v1/credit-cars")
public class AllCreditCardsRetrievalController {

    @Autowired
    private AllCreditCardsRetrievalService service;

    @GetMapping
    @ResponseStatus(value = HttpStatus.OK)
    public Flux<CreditCardResponseDto> findAll() {
        return service.findAll();
    }

    @GetMapping("{cardNumber}")
    @ResponseStatus(value = HttpStatus.OK)
    public Mono<CreditCardResponseDto> findByCardNumber(@Valid @PathVariable String cardNumber) {
        return service.findByCardNumber(cardNumber);
    }

}

File: kotlin/v2/src/main/kotlin/com/api/v2/creditcard/controllers/AllCreditCardsRetrievalController.kt
package com.api.v2.creditcard.controllers

import com.api.v2.creditcard.dtos.CreditCardResponseDto
import com.api.v2.creditcard.services.AllCreditCardsRetrievalService
import kotlinx.coroutines.flow.Flow
import org.springframework.beans.factory.annotation.Autowired
import org.springframework.http.HttpStatus
import org.springframework.web.bind.annotation.*

@RestController
@RequestMapping("api/v2/credit-cards")
class AllCreditCardsRetrievalController {

    @Autowired
    private lateinit var service: AllCreditCardsRetrievalService

    @GetMapping("{cardNumber}")
    @ResponseStatus(value = HttpStatus.OK)
    suspend fun findByCardNumber(@PathVariable cardNumber: String): CreditCardResponseDto {
        return service.findByCardNumber(cardNumber)
    }

    @GetMapping
    @ResponseStatus(value = HttpStatus.OK)
    suspend fun findAll(): Flow<CreditCardResponseDto> {
        return service.findAll()
    }

}